### PR TITLE
Eegsim

### PR DESCRIPTION
**Pull request by abuebola**
Jansen's single column model
def simulateEEGJansen(duration=10, samplingFreq=100, C1=135, callback=None):
user can specify: duration (in seconds), sampling frequency (hertz), and a callback function
callback is a callback function used to provide the UI with progress updates in %, because solving ODEs taking some time.

return value is a [time, value] tuple.
ex: 
```
eeg = simulateEEGJansen(duration=60, samplingFreq=200)
time = eeg[0]
samples = eeg[1]
```

### CODE REVIEW

File: core/simulations/eeg/jansen.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# Minimum duration: 1 sec
+# Input parameters: duration, sampling frequency, C1, callback function
+# Output: tuple with time and y arrays
+
+from numpy import linspace, random, exp, asarray, dtype, around
+from scipy.integrate import ode
+from scipy.interpolate import interp1d
+
+
+def simulateEEGJansen(duration=10, samplingFreq=100, C1=135, callback=None):
```
Comment (niroigen) [inline core/simulations/eeg/jansen.py]: I'm not sure if I've ever seen this before but you made the function simulateEEGJansen contain functions inside of it? Why not make it a class?

Comment (abuebola) [reply core/simulations/eeg/jansen.py]: why make it a class?

Comment (niroigen) [reply core/simulations/eeg/jansen.py]: Don't really have a reason other than because there are multiple functions within a function.

Comment (abuebola) [reply core/simulations/eeg/jansen.py]: i guess we can take out the two inner functions func and sigmoid and put them at the same level as simulateEEGJansen() if that makes a difference

File: core/simulations/eeg/jansen.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    def sigmoid(v0, e0, r, v):
+        sigm = (2 * e0) / (1 + exp(r * (v0 - v)))
+        return sigm
+
+    def func(t, y, noiseinterp, A, B, a, b, v0, C1, C2, C3, C4, r, e0):
```
Comment (niroigen) [inline core/simulations/eeg/jansen.py]: You're changing your naming conventions, stick with one. Here it's `noiseinterp` and then you use something else like `samplingFreq`

File: core/simulations/eeg/jansen.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    a = 100
+    b = 50
+    v0 = 6
+    C2 = 0.8 * C1
+    C3 = 0.25 * C1
+    C4 = 0.25 * C1
+    r = 0.56
+    e0 = 2.5
+
+    timeArray = linspace(0, duration+1, num=1 + samplingFreq * (duration+1))
```
Comment (niroigen) [inline core/simulations/eeg/jansen.py]: Change to `time_array`

File: core/simulations/eeg/jansen.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    noise = random.randint(120, high=320+1, size=timeArray.size) #uniformly distributed white noise
+    noiseinterp = interp1d(timeArray, noise, fill_value="extrapolate")  #noise interp function
```
Comment (niroigen) [inline core/simulations/eeg/jansen.py]: `noise_interp`

File: core/simulations/eeg/jansen.py (hunk 1 above)
Comment (abuebola) [reply core/simulations/eeg/jansen.py]: k I've removed the two nested functions and put them at the same level as simulateEEGJansen(), you're right that's better
though I don't know why would we make it a class

Comment (niroigen) [reply core/simulations/eeg/jansen.py]: Yeahh that works

File: core/simulations/eeg/jansen.py (hunk 2 above)
Comment (abuebola) [reply core/simulations/eeg/jansen.py]: I agree, it's that damn pycharm. I'll make sure to stick with one.

**Pull request merged**